lib/rouge/lexers/xojo.rb

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Xojo < RegexLexer
+      title "Xojo"
+      desc "Xojo"
+      tag 'xojo'
+      aliases 'realbasic'
+      filenames '*.xojo_code', '*.xojo_window'
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          addhandler aggregates array asc assigns attributes begin break
+          byref byval call case catch class const continue char ctype declare
+          delegate dim do downto each else elseif end endif enum event exception
+          exit extends false finally for function global goto if
+          implements inherits interface lib loop mod module
+          new next nil object of optional paramarray
+          private property protected public raise raiseevent rect redim
+          removehandler return select shared soft static step sub super
+          then to true try until using uend uhile
+        )
+      end
+
+      def self.keywords_type
+        @keywords_type ||= Set.new %w(
+          boolean cfstringref cgfloat cstring curency date double int8 int16
+          int32 int64 integer ostype pstring ptr short single
+          single string structure variant uinteger uint8 uint16 uint32 uint64
+          ushort windowptr wstring
+        )
+      end
+
+      def self.operator_words
+        @operator_words ||= Set.new %w(
+          addressof and as in is isa mod not or xor
+        )
+      end
+
+      def self.builtins
+        @builtins ||= Set.new %w(
+          app self me
+        )
+      end
+
+      id = /[a-z_]\w*/i
+      upper_id = /[A-Z]\w*/
+
+      state :whitespace do
+        rule /\s+/, Text
+        rule /\n/, Text, :bol
+        rule /rem\b.*?$/i, Comment::Single
+        rule /\/\/.*$/, Comment::Single
+        rule /\#tag Note.*\#tag EndNote/m, Comment::Doc
+      end
+
+      state :bol do
```

Comment:
Two points:

1. This state should probably have a different name. The `:bol` state is typically used by lexers for a state that may occur at the beginning, and only at the beginning, of a file. For example, preprocessor commands in Swift:
  https://github.com/rouge-ruby/rouge/blob/04ce735a0459d1f5ed7301295a7af9b0823e3e64/lib/rouge/lexers/swift.rb#L35-L42

2. Should the `Name::Attribute` rule even be here? It feels like something that should be part of the `:root` state.